File: frontend/src/types/pulse.ts
import type { ContentModule } from "./session";

export type ReportCategory = "People & Culture" | "Meetings" | "Wellness" | "Compliance";
export type ReportStatus = "focus" | "unread" | "archived" | "read";

export interface PulseReport {
  id: string;
  title: string;
  agent_name: string;
  agent_id: string;
  category: ReportCategory;
  status: ReportStatus;
  preview: string;
  markdown: string;
  modules: ContentModule[] | null;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/pulse.py
from typing import Optional

from pydantic import BaseModel

from app.schemas.session import ContentModule


class PulseReportInfo(BaseModel):
    id: str
    title: str
    agent_name: str
    agent_id: str = ""
    category: str
    status: str
    preview: str
    markdown: str
    modules: Optional[list[ContentModule]] = None
    created_at: str
    updated_at: str


class PulseReportListResponse(BaseModel):
    reports: list[PulseReportInfo]
    total: int


class PulseReportStatusUpdate(BaseModel):
    status: str
